omega/omega-transaction/src/main/java/org/apache/servicecomb/pack/omega/transaction/ForwardRecovery.java

```diff
@@ -66,6 +66,7 @@ public Object applyTo(ProceedingJoinPoint joinPoint, Compensable compensable, Co
         }
       }
     } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
```

Comment:
If this fails, the exception thrown is not as expected. 
This may cause the unit test ForwardRecoveryTest.throwExceptionWhenRetryReachesMaximum  to fail

Reply:
I read the code and found that the InterruptedException is thrown by the sleep on line 65 (which is not shown here).
The most likely reason why sleep is interrupted is some one shutdown the application. So I think it's safe to log it and rethrow it.
The previous code version just swallow the InterruptedException that prevent the application shutdown, that's not good, that's why we need to recover the interrupt state or rethrow InterruptedException.
Here is the node when I study interruption, hope it will be helpful: https://chanjarster.github.io/post/concurrent-programming/cancellation-and-shutdown/

Reply:
@chanjarster very nice summary!
Catching the InterruptedException and throw a new RuntimeException, We didn't swallow it. So I don't think I need to call Thread.currentThread().interrupt()

Reply:
When InterruptedException is caught, the Thread's interruption state is cleared, so technically speaking, even you throw a RuntimeException, the interruption is cleared, so I say "swallowed".
Thread.currentThread().interrupt() can recover the interruption state

Reply:
I use the method mentioned [here](https://chanjarster.github.io/post/java-remote-debug/) and [here](https://maven.apache.org/surefire/maven-surefire-plugin/examples/debugging.html) to debug the code when `mvn clean install`.
Weird things happens, the execution goes into ForwardRecovery#L69 which is not possible:
```java
    } catch (InterruptedException e) {
>L69  String errorMessage = "Failed to handle tx because it is interrupted, global tx id: " + context.globalTxId()
          + ", local tx id: " + context.localTxId() + ", method: " + method.toString();
      LOG.error(errorMessage);
      interceptor.onError(parentTxId, compensationMethodSignature(joinPoint, compensable, method), e);
      throw new OmegaException(errorMessage);
    }
```